elpis/engines/kaldi/objects/model.py

```diff
@@ -216,70 +206,36 @@ def prepare_for_training():
-            if run_log_path.is_file():
-                os.remove(run_log_path)
-            run(f"touch {run_log_path};")
-
-            # Organise stage logs in a dir
-            train_log_dir = self.path.joinpath('train-logs')
-            if train_log_dir.exists():
-                shutil.rmtree(train_log_dir)
-            os.mkdir(train_log_dir)
-
             self.config['stage_count'] = 0
             stages = os.listdir(local_kaldi_path.joinpath('stages'))
 
             for stage in sorted(stages):
                 print(f"Stage {stage} starting")
-                self.stage_status = (stage, 'in-progress', '', 'starting')
+                self.stage_status = (stage, 'in-progress')
                 self.config['current_stage'] = stage
 
-                # Create log file
-                stage_log_path = train_log_dir.joinpath(f"stage_{self.config['stage_count']}.log")
-                with open(stage_log_path, 'w+'):
-                    pass
-
-                #  Manipulate stage templates with user-defined settings
-                # TODO replace with jinja templates or something similar
+                # TODO update stage templates with jinja templates or something similar
                 with open(local_kaldi_path.joinpath('stages').joinpath(stage), 'r') as file :
                     filedata = file.read()
                 # Add settings to replace here
-                filedata = filedata.replace('lm_order=1', f'lm_order={self.ngram}')
+                filedata = filedata.replace('lm_order=1', f'lm_order={self.settings["ngram"]}')
                 with open(local_kaldi_path.joinpath('stages').joinpath(stage), 'w') as file:
                     file.write(filedata)
 
                 # Run the command, log output. Also redirect Kaldi sterr output to log. These are often not errors :-(
                 try:
-                    stage_process = run(f"cd {local_kaldi_path}; stages/{stage} &> {stage_log_path}")
-                    print('done')
+                    run(f"cd {local_kaldi_path}; stages/{stage} >> {self.run_log_path}")
                     print(f"Stage {stage} complete")
-                    with open(stage_log_path, 'r') as file:
-                        stage_log = file.read()
-                    print(f"Stage {stage} log", stage_log)
-                    self.stage_status = (stage, 'complete', '', stage_log)
+                    self.stage_status = (stage, 'complete')
                     self.config['stage_count'] = self.config['stage_count'] + 1
                 except CalledProcessError as error:
-                    with open(stage_log_path, 'a+') as file:
+                    with open(self.run_log_path, 'a+') as file:
                         print('stderr', error.stderr, file=file)
                         print('failed', file=file)
                     print(f"Stage {stage} failed")
                     self.stage_status = (stage, 'failed', '', 'LOG-C')
```

Comment:
I think this will cause an error after the refactor which removes the final two arguments from this setter.

Reply:
well spotted. I've removed the two final.
`self.stage_status = (stage, 'failed')`